### PR TITLE
[FEAT] Use single header yaml parser `mini-yaml`

### PR DESCRIPTION
**Pull request by acezen**
## Proposed changes
Use a single header yaml library like mini-yaml to avoid initialize submodule recursive

## Types of changes

What types of changes does your code introduce to GraphAr?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
Fix issue #141

### CODE REVIEW

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

File: cpp/test/test_info.cc [hunk 1]
```diff
@@ -108,6 +110,13 @@ TEST_CASE("test_vertex_info") {
   REQUIRE(v_info.AddPropertyGroup(pg2).IsInvalidOperation());
   REQUIRE(v_info.GetPropertyGroups().size() == 1);
 
+  GAR_NAMESPACE::Property p2;
+  p2.name = "name";
+  p2.type = GAR_NAMESPACE::DataType(GAR_NAMESPACE::Type::STRING);
+  p2.is_primary = false;
+  GAR_NAMESPACE::PropertyGroup pg3({p2}, GAR_NAMESPACE::FileType::CSV);
+  REQUIRE(v_info.AddPropertyGroup(pg3).ok());
+
```
Comment (lixueclaire) [inline cpp/test/test_info.cc:119]: Can you add the missing tests (TODO items) in this file? They are already implemented in [construct_info_example.cc](https://github.com/alibaba/GraphAr/blob/main/cpp/examples/construct_info_example.cc)

Comment (acezen) [reply cpp/test/test_info.cc:119]: OK, but this is not related to this pull request, I will address the issue with another pull request

**Review by lixueclaire — APPROVED** [review]
LGTM~

**Review by sighingnow — APPROVED** [review]

**Pull request merged**